### PR TITLE
Ports/tar: Update to version 1.35

### PR DESCRIPTION
**Pull request by supercomputer7**
This version requires the gettext port as a dependency. We also disable the unused-parameter warning during compilation, as it's not very interesting and will actually make the build to fail.

### CODE REVIEW

File: Ports/tar/patches/0001-Disable-Wunused-parameter-warnings.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+diff --git a/m4/gnulib-common.m4 b/m4/gnulib-common.m4
+index 2426eb639852f5e7d07e1c04e609cb174f2e3498..4a528b8dfa0c5b5f76d3125fbe7e21215fa5e7be 100644
+--- a/m4/gnulib-common.m4
++++ b/m4/gnulib-common.m4
+@@ -1070,7 +1070,7 @@ AC_DEFUN([gl_CC_GNULIB_WARNINGS],
+       -Wno-sign-compare
+       -Wno-undef
+       -Wno-unused-function
+-      -Wno-unused-parameter
```
Comment (BertalanD) [inline Ports/tar/patches/0001-Disable-Wunused-parameter-warnings.patch:18]: I'm a bit confused. Doesn't this line already disable the `-Wunused-parameter` warning already?

Comment (supercomputer7) [reply Ports/tar/patches/0001-Disable-Wunused-parameter-warnings.patch:18]: Should be OK now :)

Comment (timschumi) [reply Ports/tar/patches/0001-Disable-Wunused-parameter-warnings.patch:18]: This is confusing for two reasons actually. First obviously being that we are actually removing the option that is supposed to suppress those errors here. The second is that we are changing an input file for `autoconf` here, so nothing that we do here would affect the build without running `autoconf` first.

Comment (supercomputer7) [reply Ports/tar/patches/0001-Disable-Wunused-parameter-warnings.patch:18]: I now ensured that in addition to just removing the `-Wno-unused-parameter` flag, we do it in the `configure` file :)

**Comment by timschumi** [comment]
For documentation purposes: Support for external libintl and libiconv libraries in tar 1.35 is [broken](https://savannah.gnu.org/bugs/?64441) and the [fix](https://git.savannah.gnu.org/cgit/tar.git/commit/?id=8632df398b2f548465ebe68b8f494c0d6f8d913d) seems to just have been put onto `master` as usual. Patching the Makefile.am files is not practical for us, so we'll have to override it manually or skip this release entirely.

File: Ports/tar/package.sh [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
-version='1.34'
+version='1.35'
 useconfigure='true'
 use_fresh_config_sub='true'
 config_sub_paths=('build-aux/config.sub')
 files=(
-    "https://ftpmirror.gnu.org/gnu/tar/tar-${version}.tar.gz 03d908cf5768cfe6b7ad588c921c6ed21acabfb2b79b788d1330453507647aed"
+    "https://ftpmirror.gnu.org/gnu/tar/tar-${version}.tar.gz 14d55e32063ea9526e057fbf35fcabd53378e769787eff7919c3755b02d2b57e"
 )
+depends=("gettext")
```
Comment (timschumi) [inline Ports/tar/package.sh:10]:
```suggestion
depends=(
    'gettext'
)
```

File: Ports/tar/package.sh [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 configopts=(
     "--without-selinux"
     "--without-posix-acls"
     "--without-xattrs"
+    "--disable-nls"
+    "--disable-libiconv"
```
Comment (timschumi) [inline Ports/tar/package.sh:16]: Why are we adding these now?

File: Ports/tar/patches/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Patches for tar on SerenityOS
+
+## `0001-Fix-savannah-bug-64441.patch`
+
+Release 1.35 has a bug in which it doesn't include libintl and libiconv properly.
+Add libiconv libraries to the build system.
```
Comment (timschumi) [inline Ports/tar/patches/ReadMe.md:6]: This doesn't match the patch that the README is generated from. We should either modify the patch directly to say that or just leave the autogenerated README as-is.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/tar/package.sh (hunk 3 above)
Comment (supercomputer7) [reply Ports/tar/package.sh:16]: Just a test, I forgot about these :)

**Review by timschumi — APPROVED** [review]

**Pull request merged**